### PR TITLE
Enable go-tests in Travis

### PR DESCRIPTION
**Pull request by mitkus**
For https://github.com/lightbend/es-backend/issues/562

This enables new go e2e tests to run automatically on Travis CI, alongside the old bash tests.

List of changes:
- Set Go version in Travis CI environment to 1.11.x
- Renamed `go-tests` module and directory to `gotests`
- Improved `util.Cmd()` failure reporting
- Added `kube.DeletePvc()` function
- Fixed failure to install console in test environment due to existing PVCs
- Disabled the most flaky tests

### CODE REVIEW

**Comment by marcoderama** [comment]
Is this still a WIP?

**Comment by mitkus** [comment]
The test suite runs and successfully passes in Travis, should be good for review now.

File: .travis.yml [hunk 1]
```diff
@@ -29,6 +31,12 @@ jobs:
       - scripts/setup-openshift.sh || travis_terminate 1
       - make -C enterprise-suite openshift-backend-tests
 
+  - name: Backend E2E Go Tests - Minikube
+    script:
+    - scripts/setup-minikube-for-linux.sh || travis_terminate 1
+    - go get github.com/onsi/ginkgo/ginkgo || travis_terminate 1
+    - env GO111MODULE=on make -C enterprise-suite go-tests
```
Comment (jsravn) [inline .travis.yml]: Is `env` needed?

File: enterprise-suite/gotests/util/util.go [hunk 2]
```diff
@@ -124,14 +124,20 @@ func (cb *CmdBuilder) Run() (int, error) {
...
 		panic("unable to execute command")
 	} else {
-		var stdoutWriters []io.Writer
-		var stderrWriters []io.Writer
+		// These keep all stdout/stderr capture destinations
+		stdoutWriters := []io.Writer{&errStdoutBuilder}
+		stderrWriters := []io.Writer{&errStderrBuilder}
 
-		// Add string.Builder outputs
+		// Add external string.Builder outputs
```
Comment (jsravn) [inline enterprise-suite/gotests/util/util.go]: For the io.Copys below - are you sure those always return? In the past I ran them in goroutines, then use Wait (potentially with a timeout).

File: enterprise-suite/gotests/util/util.go [hunk 3]
```diff
@@ -124,14 +124,20 @@ func (cb *CmdBuilder) Run() (int, error) {
 		panic("unable to get command stderr pipe")
 	}
 
+	// Always capture stdout and stderr to these string builders so that
+	// we can print them out in case of errors
+	var errStdoutBuilder strings.Builder
+	var errStderrBuilder strings.Builder
```
Comment (jsravn) [inline enterprise-suite/gotests/util/util.go]: Can you use the `GinkgoWriter`? It will auto print out only if a test fails.

Comment (mitkus) [reply enterprise-suite/gotests/util/util.go]: Switched to GinkgoWriter here

File: .travis.yml (hunk 1 above)
Comment (mitkus) [reply .travis.yml]: Not really, removed

**Pull request merged**